pkg/operator/operator.go

```diff
@@ -811,6 +812,38 @@ func (op *Reporting) newPrometheusConn(promConfig promapi.Config) (prom.API, err
 	return prom.NewAPI(client), nil
 }
 
+func (op *Reporting) handleExpiredReport(report *metering.Report, now time.Time) error {
+	// check if the Report is being deleted already
+	if report.DeletionTimestamp != nil {
+		return fmt.Errorf("report was already marked for deletion")
+	}
+	// check if the Report is past its retention time
+	if reportExpired := isReportExpired(op.logger, report, now); !reportExpired {
+		op.logger.Debugf("the %s Report in the %s namespace has not reached the expiration date", report.Name, report.Namespace)
+		return nil
+	}
+	// check if the Report is used by any other Report or ReportQuery, if not delete it
+	if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+		newDeletionTimestamp := metav1.Now()
+		report.SetDeletionTimestamp(&newDeletionTimestamp)
+		err := op.meteringClient.MeteringV1().Reports(report.Namespace).
+			Delete(context.TODO(), report.Name, metav1.DeleteOptions{})
+		if apierrors.IsNotFound(err) {
+			op.logger.Infof("report: %s, not deleted because it was not found at time delete attempted", report.Name)
+			return nil
+		}
+		if err != nil {
+			err = fmt.Errorf("should have deleted report: %s, because expired, but failed to delete", report.Name)
+			return err
+		}
+		op.logger.Infof("deleted report: %s, because expired", report.Name)
+		return nil
+	} else {
+		op.logger.Warnf("report: %s, would be deleted because expired, but is depended on", report.Name)
+		return nil
+	}
```

Comment:
This is a redundent else clause
```suggestion
	op.logger.Warnf("report: %s, would be deleted because expired, but is depended on", report.Name)
	return nil
```

Reply:
fair. I added a comment to clarify the logic at that point though w/out the else.